CONTRIBUTING.md

```diff
@@ -1,61 +1,520 @@
+"_how_" is an opinionated implementation that you can override with
+"escape hatches." This will allow you to have higher velocity and build
+better applications by focusing less on implementation choices.
+
+Secondly, Amplify is a manifestation of
+[The Rule of Least Power](https://en.wikipedia.org/wiki/Rule_of_least_power)
+when developing against AWS. This means it encourages architectural and
+programmatic best practices that enable you to start quickly.  This
+shows by encouraging certain services (API Gateway usage vs. direct
+DynamoDB interaction) or certain connection patterns (Circuit breaker,
+retry counts and throttle up/down).
+
+__Opinionated implementations__: There are many ways to interface with AWS
+Services. Certain service interactions are favored over others. For instance, if
+sending and receiving JSON, we would prefer an API Gateway endpoint to other
+mechanisms. Amplify will programmatically help optimize for cost and
+performance through library decisions.
+
+__Declarative actions__: Amplify will provide you a reference to a generic client
+object and offer an ability to perform common actions: "RegisterUser", "Login",
+"SendObject", "UpdateObject", "StreamData". By default, you should not need to
+worry about AWS-service-specific API operations like `putItem()` with a unique
+hash -- or even which HTTP verbs to use.
+
+__Cascading service interactions__: Certain actions in a declarative style can have
+overlapping or ambiguous AWS Service implementations. With an opinionated
+implementation we can decide which Services are "primary" and which are
+"secondary" depending on what is configured. For instance sending an image will
+prefer S3 over API Gateway..
+
+__Simple, standard data objects__: Sending & Receiving data to AWS Services can
+have many parameters, which tend to show up in the SDKs. These are abstracted
+and inferred, where possible, with simple JSON that the implementation can
+reason about. Standard parameters (bucket names, stream names, partition keys,
+etc.) that are part of the implementation are extracted from a simplified
+configuration file and dynamically generated/updated in order to further allow
+focus on state and data types only.
+
+### Amplify Design
+As more and more modules are introduced to AWS Amplify, there is a clear
+necessity to modularize the library into smaller pieces so that users
+could avoid importing unnecessary parts into their app. The goal of this
+design is to make AWS Amplify modularized and also keep it backward
+compatible to avoid breaking changes.
+
+Modular import prevents unnecessary code dependencies are included with the
+app, and thus decreases the bundle size and enables adding new functionality
+without the risk of introducing errors related to the unused code.
+
+Amplify has established the concepts of categories and plugins. A
+__category__ is a collection of api calls that are exposed to the client
```

Comment:
A category is a grouping of functional use cases that a developer would add to a mobile or web application such as "Auth" or "Analytics". The API calls exposed in that category are declarative in nature for those use case operations.

Reply:
(Moving to a later PR.)